src/Mvc/test/WebSites/BasicWebSite/Startup.cs

```diff
@@ -48,6 +52,16 @@ public void ConfigureServices(IServiceCollection services)
             services.AddTransient<ServiceActionFilter>();
             services.AddScoped<TestResponseGenerator>();
             services.AddSingleton<IActionContextAccessor, ActionContextAccessor>();
+            services.TryAddSingleton(CreateWeatherForecastService());
+        }
+
+        private WeatherForecastService CreateWeatherForecastService()
+        {
+            // For manual debug only,
+            // This needs to be changed to match the site host
+            var client = new HttpClient();
+            client.BaseAddress = new Uri("http://localhost:51733/");
```

Comment:
Instead of newing up the `HttpClient` manually, can you get an `HttpClient` from the DI container? If you do, its base address should already be populated correctly.

Reply:
Actually I'm not 100% certain about this. That's true for the web assembly DI container. Might not be for the non-wasm one.

Reply:
That's not true in asp.net core.